http/src/main/java/io/crate/plugin/PipelineRegistry.java

```diff
@@ -73,6 +92,7 @@ public void registerItems(ChannelPipeline pipeline) {
         for (PipelineRegistry.ChannelPipelineItem item : addBeforeList) {
             pipeline.addBefore(item.base, item.name, item.handlerFactory.get());
         }
+        sslHandler.addToPipeline(pipeline);
```

Comment:
Not sure i like that this is bi-directional. Wouldn't it be possible to remove any SslHandler knowledge from the PipelineRegistry? I think we should have some service in `ssl-impl` that creates the context and then simply uses the PipelineRegistry to add the SslHandler. This would also remove the need for different `HttpsHandler` implementations